eureka-core/src/main/java/com/netflix/eureka/registry/rule/DownOrStartingRule.java

```diff
@@ -0,0 +1,39 @@
+package com.netflix.eureka.registry.rule;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.eureka.lease.Lease;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * If the instance is STARTING or DOWN then we return that.
+ * This rule has precedence over the {@link OverrideExistsRule}.
```

Comment:
This comment is misleading. From the rule interface design, each individual rule should not have understanding of where it applies in relation to other rules, and it is up to the definer at the top level to define them in the appropriate hierarchy. Perhaps move this comment to the *registry.java's?